### PR TITLE
[DCA] Convert commands to Fx apps

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

Convert Datadog Cluster Agent subcommands to Fx apps, as part of the effort of componentizing all of the Datadog Agent commands.

### Describe how to test/QA your changes

Ensure that the following commands still work as previous versions:

* check
* clusterchecks
* completion
* compliance
* config
* configcheck
* diagnose
* flare
* health
* help
* metamap
* secret-helper
* start
* status
* telemetry
* version

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM for the components part. 
Note: I did check the code behave the same as before.

File: cmd/cluster-agent/subcommands/compliance/command.go [hunk 1]
```diff
@@ -6,34 +6,30 @@
 //go:build !windows && kubeapiserver
 // +build !windows,kubeapiserver
 
-package app
+// Package compliance implements 'cluster-agent compliance'.
+package version
```
Comment (clamoriniere) [inline cmd/cluster-agent/subcommands/compliance/command.go:10]: is it the right package name? the folder name and the comment said it should be `compliance` not `version`

File: cmd/cluster-agent/subcommands/compliance/command.go [hunk 2]
```diff
@@ -6,34 +6,30 @@
...
-		ConfigParams: config.NewParams(
-			"",
-			config.WithConfFilePath(confPath),
-			config.WithConfigName("datadog-cluster"),
-		),
-		LogParams: log.LogForOneShot(string(loggerName), "off", true),
-	}
+	complianceCmd.AddCommand(check.Commands(core.BundleParams{
+		ConfigParams: config.NewClusterAgentParams(""),
+		LogParams:    log.LogForOneShot("CLUSTER", "off", true),
```
Comment (clamoriniere) [inline cmd/cluster-agent/subcommands/compliance/command.go:31]: can we reuse the `const logerName = "CLUSTER"` defined here: https://github.com/DataDog/datadog-agent/pull/15009/files#diff-743e6399c0f80ca0502958e39dbfd259e8df0889ff5ae1a04717dfed3b76e09aR19

File: cmd/cluster-agent/subcommands/config/command.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/cli/subcommands/config"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/settings"
+	settingshttp "github.com/DataDog/datadog-agent/pkg/config/settings/http"
+
+	"github.com/spf13/cobra"
+)
+
+const loggerName = "CLUSTER"
```
Comment (clamoriniere) [inline cmd/cluster-agent/subcommands/config/command.go:25]: loggerName seems to be defined in several places, can we create a dedicated package for const values? and reuse the same const variable everywhere?

File: cmd/cluster-agent/subcommands/diagnose/command.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "diagnose",
+		Short: "Execute some connectivity diagnosis on your system",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return fxutil.OneShot(run,
+				fx.Supply(core.BundleParams{
+					ConfigParams: config.NewClusterAgentParams(globalParams.ConfFilePath),
+					LogParams:    log.LogForOneShot("CLUSTER", "off", true),
```
Comment (clamoriniere) [inline cmd/cluster-agent/subcommands/diagnose/command.go:34]: same comment about `CLUSTER`

**Review by clamoriniere — COMMENTED** [review]
Awesome work 👏 

few minors comments

File: cmd/cluster-agent/subcommands/compliance/command.go (hunk 1 above)
Comment (juliogreff) [reply cmd/cluster-agent/subcommands/compliance/command.go:10]: You are correct. Fixed 👍

**Review by clamoriniere — APPROVED** [review]

**Review by sarah-witt — APPROVED** [review]
Looks good! This will be helpful for moving the cloud foundry cluster agent to fx!

**Pull request merged**